src/main/java/hudson/plugins/audit_trail/AuditTrailPlugin.java

```diff
@@ -23,159 +23,146 @@
 import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
 /**
  * Keep audit trail of particular Jenkins operations, such as configuring jobs.
+ *
  * @author Alan Harder
+ * @author Pierre Beitz
  */
-public class AuditTrailPlugin extends Plugin {
+@Symbol("audit-trail")
+@Extension
+public class AuditTrailPlugin extends GlobalConfiguration {
+
+    private static final Logger LOGGER = Logger.getLogger(AuditTrailPlugin.class.getName());
     private String pattern = ".*/(?:configSubmit|doDelete|postBuildResult|enable|disable|"
-      + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
-      + "cancelQuietDown|quietDown|restart|exit|safeExit)";
+            + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
+            + "cancelQuietDown|quietDown|restart|exit|safeExit)";
     private boolean logBuildCause = true;
 
-    private List<AuditLogger> loggers = new ArrayList<AuditLogger>();
+    private List<AuditLogger> loggers = new ArrayList<>();
 
     private transient boolean started;
 
     private transient String log;
-    private transient int limit = 1, count = 1;
 
     public String getPattern() { return pattern; }
     public boolean getLogBuildCause() { return logBuildCause; }
     public List<AuditLogger> getLoggers() { return loggers; }
 
-    @Override public void start() throws Exception {
-        // Set a default value; will be overridden by load() once customized:
+    public AuditTrailPlugin() {
         load();
-        applySettings();
-
-        // Add Filter to watch all requests and log matching ones
-        PluginServletFilter.addFilter(new AuditTrailFilter(this));
     }
 
-    @Override public void configure(StaplerRequest req, JSONObject formData)
-            throws IOException, ServletException, FormException {
-        pattern = formData.optString("pattern");
-        logBuildCause = formData.optBoolean("logBuildCause", true);
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
         // readResolve makes sure loggers is initialized, so it should never be null.
+        // TODO this should probably be moved somewhere else
         loggers.forEach(AuditLogger::cleanUp);
-        loggers = Descriptor.newInstancesFromHeteroList(
-                req, formData, "loggers", getLoggerDescriptors());
-        save();
+        try (BulkChange bc = new BulkChange(this)) {
+            req.bindJSON(this, formData);
+            bc.commit();
+        } catch (IOException e) {
+            throw new FormException("Failed to apply configuration", e, null);
+        }
         applySettings();
+        return true;
+    }
+
+    @DataBoundSetter
+    public void setPattern(String pattern) {
+        this.pattern = Optional.ofNullable(pattern).orElse("");
```

Comment:
either `fixNull` or `fixEmpty` would be simpler,  I  prefer `fixEmpty` normally empty string is just a hack

Reply:
mmmhh, I'd rather use native features instead of some custom functions. I understand this was created pre jdk8 but now that jdk8 is used I'm not sure there is a strong point against using the native way. Am I missing something?

Reply:
It’s a lot more verbose and you’re always creating an object for no reason.

But each to their own

Reply:
Thanks for the explanation, that makes sense 👍